plugins/circleci-npm/src/index.ts

```diff
@@ -1,15 +1,18 @@
-import CircleCiConfigHook from '@dotcom-tool-kit/circleci/lib/circleci-config'
+import CircleCiConfigHook, { generateConfigWithJob } from '@dotcom-tool-kit/circleci/lib/circleci-config'
 
 class PublishHook extends CircleCiConfigHook {
-  job = 'tool-kit/publish'
-  jobOptions = {
-    context: 'npm-publish-token',
+  static job = 'tool-kit/publish'
+  config = generateConfigWithJob({
+    name: PublishHook.job,
     requires: ['tool-kit/test'],
-    filters: {
-      branches: { ignore: '/.*/' }
+    addToNightly: false,
+    additionalFields: {
+      context: 'npm-publish-token',
+      filters: {
+        branches: { ignore: '/.*/' }
+      }
     }
-  }
-  runOnVersionTags = true
```

Comment:
**praise:** nice getting rid of this! Not having to evaluate this in all the places feels cleaners and more simple 😊